app/services/setting_registry.rb

```diff
@@ -1,8 +1,23 @@
 class SettingRegistry
   include Singleton
+  include Enumerable
 
-  def initialize
-    @settings = {}
+  def self.subset_registry(subset)
+    new(subset)
+  end
+
+  def search_for(query)
+    return self if query.blank?
```

Comment:
Is returning self (SettingRegistry) really correct? I'd expect this would return a setting model rather than registry.

Reply:
This returns self, that is (acts as) a collection, thus it mimics what `ActiveRecord.search_for()` does.